### PR TITLE
Fix crashes on yt_dlp filters (on youtube for example)

### PR DESCRIPTION
**Pull request by Miezhiko**

### CODE REVIEW

File: src/yt_utils.py [hunk 1]
```diff
@@ -26,19 +26,25 @@ def get_best(formats):
 
 
 def get_best_audio(formats):
-    filtered = list(filter_audio(formats))
+    filtered = list(filter_video(formats))
+    if not filtered:
+        filtered = list(filter_audio_video(formats))
```
Comment (jeffshee) [inline src/yt_utils.py:31]: `get_best_audio` should return audio-only. I think there is nothing wrong with the previous implementation.

**Review by jeffshee — CHANGES_REQUESTED** [review]

**Comment by jeffshee** [comment]
@Miezhiko Hi, thanks for the PR.

Please include the description of PR next time. I don't know why this fix is needed, since I had done some testing with youtube videos, and all of them work fine. It will be great if you could provide the youtube link so that I can reproduce the issue. Without that, it's hard to tell if the current PR fixes the issue correctly or not.

**Comment by Miezhiko** [comment]
@jeffshee inside formats there are (for some youtube stream I've tried) few items and all of them has either acodec and vcodec.

in methods `filter_video` or `filter_audio` it's expected that there is element with empty codec on video or audio and if there is no such item next method `max` or `min` will crash whole application in runtime because there is no check for empty collection.

File: src/yt_utils.py (hunk 1 above)
Comment (Miezhiko) [reply src/yt_utils.py:31]: honestly, didn't checked where `get_best_audio` used and why this method needed but previous implementation will at least need check for empty filtered list before `min` or `max`

Comment (jeffshee) [reply src/yt_utils.py:31]: I agree with you. Maybe you should revert the line 29, but keep the changes that you had made on line 30 and 31.

**Comment by jeffshee** [comment]
@Miezhiko The reason I didn't use the result of `filter_audio_video` is, the url retrieved always links to low res video for YouTube based on my tests. It seems YouTube always stream high quality video and audio separately, and `filter_video`, `filter_audio` are the filters to get both the links we need.

However you are right with the empty collection check, which I overlooked. There might be some edge cases (live stream maybe?) that we have no choice but to use the `filter_audio_video` as the fallback option.

Please ask me if you have any question 😃

File: src/yt_utils.py (hunk 1 above)
Comment (Miezhiko) [reply src/yt_utils.py:31]: ahhh line 29 change is just typo, sorry

Comment (jeffshee) [reply src/yt_utils.py]: Thanks

**Review by jeffshee — APPROVED** [review]

**Pull request merged**